src/core/dataProviders/authProvider.js

```diff
@@ -35,13 +35,28 @@ const FetchLogin = (resolve, reject) => {
         });
 };
 
+const FetchLogout = (resolve, reject) => {
+    const urlLogout = domainName + '/api/logout';
+    fetch(urlLogout, options)
+        .then(res => res.json())
+        .then(response => {
+            const redirectUrl = get(response, 'redirectURL', '');
+            return resolve(redirectUrl);
+        });
+};
+
 export default async (type, params) => {
+
     if (type === AUTH_LOGOUT) {
-        localStorage.removeItem('token');
-        localStorage.removeItem('userId');
-        localStorage.removeItem('given_name');
-        localStorage.removeItem('family_name');
-        localStorage.removeItem('role');
+        if (localStorage.getItem('userId') && token) {
+            new Promise(FetchLogout).then(redirectUrl => {
+                document.cookie = 'JSESSIONID=;';
+                localStorage.removeItem('userId');
+                localStorage.removeItem('username');
+                localStorage.removeItem('role');
+                window.location = redirectUrl;
```

Comment:
react-admin already does a redirection to the login page *before* calling the auth provider. Besides, I find it weird that you trust the API to tell you where you should redirect the user - that's the frontend's business. So you should not change the window location.

Reply:
@fzaninotto 
Yes, RA provides redirect to login page - it works!
But we need in redirect to OpenID login page. There are URL something like:
https://{domain name}/openid/interaction/3eb0f3d9-c23c-4482-8e27-eed8c07e7575 

We have domain name at frontend part, but where can we get this code: 3eb0f3d9-c23c-4482-8e27-eed8c07e7575 ? Besides this, this code is differ each time.

For this reason we use redirect URL from response with window.location.

Reply:
Right, I understand.